docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+
+  indirectVolumeMigration: false
+
+  migStorageRef:
+    name: migstorage-sample
+    namespace: openshift-migration
+
+  # [!] Change namespaces to adjust which OpenShift namespaces should be migrated from source to destination cluster
+  namespaces:
+  - nginx-example
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+```
+
+- spec.closed: If the migration was successful for a migplan, this value can be set True indicating that after one successful migration no new migrations can be carried out for this migplan.
+
+- spec.destMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.hooks: Holds a reference to a MigHook along with the desired phase to run it in.
+    -   spec.hooks.executionNamespace: Holds the name of the namespace where hooks should be implemented.
+    -   spec.hooks.phase: Indicates the phase when the hooks will be executed. Acceptable values are: PreBackup, PostBackup, PreRestore, and PostRestore.
+    -   spec.hooks.serviceAccount: Holds the name of the service account to be used for running hooks.
+    -   spec.hooks.reference: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.indirectImageMigration: If set True, disables direct image migrations.
+
+- spec.indirectVolumeMigration: If set True, disables direct volume migrations.
+
+- spec.migStorageRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.namespaces: A list of string of all the namespaces to be included in migration.
+
+- spec.persistentVolumes: Holds list of all the persistent volumes found for the namespaces included in migplan. Each entry in the list is a persistent volume with the information. Name - The PV name. Capacity - The PV storage capacity. StorageClass - The PV storage class name. Supported - Lists of what is supported. Selection - Choices made from supported. PVC - Associated PVC. NFS - NFS properties. staged - A PV has been explicitly added/updated.
+
+- spec.refresh: If set True, the controller is forces to check if the migplan is in Ready state or not.
+
+- spec.srcMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+
+## MigStorages
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigStorage
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migstorage-sample
+  namespace: openshift-migration
+spec:
+  backupStorageProvider: aws
+  volumeSnapshotProvider: aws
+  
+  backupStorageConfig:
+    # [!] Change awsBucketName to contain the S3 bucket name to be used for migration
+    awsBucketName: foo
+    # [!] Change awsRegion to contain the region name (e.g. 'us-east-1') where the S3 bucket presides
+    awsRegion: foo
+    credsSecretRef:
+      namespace: openshift-config
+      name: migstorage-creds
+
+    # Optional backupStorageConfig parameters
```

Comment:
Why would these optional parameters be used? We do not mention them anywhere in our docs.